src/ykfde-disenroll

```diff
@@ -0,0 +1,149 @@
+
+if [ "$YKFDE_SLOT_CHECK" != 1 ]; then
+  printf '%s\n' "ERROR: Chosen YubiKey slot '$YKFDE_CHALLENGE_SLOT' isn't configured. Please choose slot configured for 'HMAC-SHA1 Challenge-Response' mode in '/etc/ykfde.conf'"
+  exit 1
+fi
+
+if [ -z "$YKFDE_PRINT_ONLY" ]; then
+  if [ -z "$YKFDE_LUKS_DEV" ]; then
+    echo "ERROR: Device not selected. Please select an existing device using '-d' option, see 'ykfde-disenroll -h' for help."
+    exit 1
+  fi
+  if [ ! -e "$YKFDE_LUKS_DEV" ]; then
+    printf '%s\n' "ERROR: Selected device '$YKFDE_LUKS_DEV' doesn't exist. Please select an existing device."
+    exit 1
+  fi
+
+  if [ -z "$LUKS_KEYSLOT" ]; then
+    echo "ERROR: LUKS keyslot not selected. Please select LUKS keyslot using '-s' option, see 'ykfde-disenroll -h' for help."
+    exit 1
+  fi
+
+  if [ "$(id -u)" -eq 0 ]; then
+    if ! cryptsetup isLuks "$YKFDE_LUKS_DEV"; then
+      printf '%s\n' "ERROR: Selected device '$YKFDE_LUKS_DEV' isn't a LUKS encrypted volume. Please select a valid device."
+      exit 1
+    fi
+  fi
+  printf '%s\n' "WARNING: This script will try to remove the '$LUKS_KEYSLOT' LUKS key slot on drive '$YKFDE_LUKS_DEV' . If this is not what you intended, please abort."
+fi
+
+[ -z "$YKFDE_CHALLENGE" ] && YKFDE_CHALLENGE_PASSWORD_NEEDED=1
+[ "$YKFDE_CHALLENGE_PASSWORD_NEEDED" ] && YKFDE_CHALLENGE=""
+while [ -z "$YKFDE_CHALLENGE" ]; do
+  echo " > Please provide the challenge."
+  printf "   Enter challenge: "
+  if [ "$DBG" ]; then read -r YKFDE_CHALLENGE; else read -r -s YKFDE_CHALLENGE; fi
+  YKFDE_CHALLENGE="$(printf %s "$YKFDE_CHALLENGE" | sha256sum | awk '{print $1}')"
+  # if /NOT/ DBG, we need to output \n here.
+  [ "$DBG" ] || echo
+done
+
+echo "   Remember to touch the device if necessary."
+[ "$DBG" ] && printf '%s\n' "   Running: 'ykchalresp -$YKFDE_CHALLENGE_SLOT '$YKFDE_CHALLENGE''..."
+# we have to temporary disable '-e' shell option otherwise there would bo no chance for second attempt
+set +e
+YKFDE_RESPONSE="$(ykchalresp -"$YKFDE_CHALLENGE_SLOT" "$YKFDE_CHALLENGE" 2>/dev/null | tr -d '\n')"
+set -e
+if [ -z "$YKFDE_RESPONSE" ]; then
+  sleep 3
+  echo "   YubiKey did not provide a response - Initializing second attempt, touch the device if necessary."
+  YKFDE_RESPONSE="$(ykchalresp -"$YKFDE_CHALLENGE_SLOT" "$YKFDE_CHALLENGE" | tr -d '\n')"
+fi
+
+[ "$DBG" ] && printf '%s\n' "   Received response: '$YKFDE_RESPONSE'"
+
+if [ "$YKFDE_RESPONSE" ]; then
+  if [ "$YKFDE_CHALLENGE_PASSWORD_NEEDED" ]; then
+    YKFDE_PASSPHRASE="$YKFDE_CHALLENGE$YKFDE_RESPONSE"
+  else
+    YKFDE_PASSPHRASE="$YKFDE_RESPONSE"
+  fi
+fi
+
+if [ "$YKFDE_PRINT_ONLY" ]; then
+  printf '%s\n' " > ykfde passphrase: $YKFDE_PASSPHRASE"
+  exit 0
+fi
+
+if [ "$(id -u)" -eq 0 ]; then
+  [ "$DBG" ] && printf '%s\n' " > Passing '$YKFDE_PASSPHRASE' to 'cryptsetup'"
+  [ "$DBG" ] && printf '%s\n' " > Killing with 'cryptsetup luksKillSlot $YKFDE_LUKS_DEV $LUKS_KEYSLOT'..." || echo " > Killing slot 'cryptsetup'..."
```

Comment:
```
In ykfde-disenroll line 136:
  printf %s "$YKFDE_PASSPHRASE" | cryptsetup luksKillSlot $YKFDE_LUKS_DEV $LUKS_KEYSLOT 2>&1
                                                          ^-------------^ SC2086: Double quote to prevent globbing and word splitting.
                                                                          ^-----------^ SC2086: Double quote to prevent globbing and word splitting.
```